From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Sun, 2 Jun 2024 19:44:31 +0800
Subject: [PATCH] Customize spigot loader maven repo You need to add:
 `-Dignite.maven`` in the startup parameters, such as `java
 -Dignite.maven="https://repo.maven.apache.org/maven2" -jar levelbukkit.jar`


diff --git a/src/main/java/net/minecraft/world/entity/decoration/HangingEntity.java b/src/main/java/net/minecraft/world/entity/decoration/HangingEntity.java
index eaad15a4d201356c34c1a09c7fbe5c35f76a2176..20a1033c9d7290e702acc088923794e4c4428ec5 100644
--- a/src/main/java/net/minecraft/world/entity/decoration/HangingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/HangingEntity.java
@@ -183,10 +183,8 @@ public abstract class HangingEntity extends Entity {
 
     @Override
     public boolean skipAttackInteraction(Entity attacker) {
-        if (attacker instanceof Player) {
-            Player entityhuman = (Player) attacker;
-
-            return !this.level().mayInteract(entityhuman, this.pos) ? true : this.hurt(this.damageSources().playerAttack(entityhuman), 0.0F);
+        if (attacker instanceof Player entityhuman) {
+            return !this.level().mayInteract(entityhuman, this.pos) || this.hurt(this.damageSources().playerAttack(entityhuman), 0.0F);
         } else {
             return false;
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 8d626fadcd4743b6472a2954d2b1b2ec89669814..c63a1755a81a6662fd2cc938bbe0afe5dd1473e0 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -7,7 +7,6 @@ import java.util.Arrays;
 import java.util.Calendar;
 import java.util.Date;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import joptsimple.OptionParser;
